server/events/project_locker.go

```diff
@@ -81,7 +91,9 @@ func (p *DefaultProjectLocker) TryLock(log logging.SimpleLogging, pull models.Pu
 	return &TryLockResponse{
 		LockAcquired: true,
 		UnlockFn: func() error {
-			_, err := p.Locker.Unlock(lockAttempt.LockKey)
+			// TODO monikma #8 this will be called if there was a plan error and the lock was automatically dropped;
```

Comment:
How about this `TODO`?

Reply:
Moving the discussion to https://github.com/runatlantis/atlantis/pull/2782#discussion_r1089984937